trio/_highlevel_socket.py

```diff
@@ -330,7 +335,7 @@ class SocketListener(Listener[SocketStream], metaclass=Final):
 
     """
 
-    def __init__(self, socket):
+    def __init__(self, socket: SocketType):
```

Comment:
Huh, TIL that mypy (strict, too) is fine without the explicit `-> None` bit here.

Reply:
Yeah `__init__` has a special case exception.